src/pages/index.js

```diff
@@ -100,36 +148,115 @@ function fillProfileForm() {
+    const newInfo = {
         name: data["name-input"],
         description: data["description-input"],
+    };
+    api.editProfile({
+        name: newInfo.name,
+        about: newInfo.description
+    })
+        .then(() => {
+            userInfo.setUserInfo(newInfo);
+        })
+        .catch((err) => {
+            console.log("Error updating user info: " + err);
     });
 }
 
+function openAvatarModal() {
+    avatarPopup.open();
+    formValidators["avatar-form"].toggleButtonState();
+}
+
+async function updateAvatar(event, data) {
+    const link = data["url-input"];
+
+    return api.updateAvatar(link)
+        .then(() => {
+            userInfo.setUserAvatar(link);
+        })
+        .then(() => {
+            event.target.reset();
+        })
+        .catch((err) => {
+            console.log("Error updating avatar: " + err);
+        });
+    
+}
+
 // #endregion 
 
-// #region New place modal & card creation methods
+// #region New place modal & card methods
 
 function openPlaceModal() {
     formValidators["place-form"].toggleButtonState();
     placePopup.open();
 }
 
-function addCard(event, data) {
+function openConfirmModal(event, data) {
+    deletePopup.open();
+    deletePopup.setConfirmationData({
+        cardElement: event.target.closest(".card"),
+        data: data,
+    });
+}
+
+async function addCard(event, data) {
     const cardData = {
         name: data["title-input"],
         link: data["url-input"],
     };
-    cardsSection.addItem(getCardElement(cardData));
-    event.target.reset();
+
+    return api.addCard(cardData)
+        .then((res) => {
+            cardData._id = res._id;
+            cardsSection.addItem(getCardElement(cardData));
+        })
+        .then(() => {
+            event.target.reset();
+        })
+        .catch((err) => {
+            console.log("Error adding a card: " + err);
+        });
+}
+
+async function deleteCard({cardElement, data}) {
+    return api.deleteCard(data._id)
+        .then(() => {
+            cardElement.remove();
+        })
+        .catch((err) => {
+            console.log("Error deleting a card: " + err);
+        });
+}
+
+async function likeCard(id) {
+    api.likeCard(id)
+        .catch((err) => {
+            console.log("Error liking a card: " + err);
+    });
+}
+
+async function unlikeCard(id) {
+    api.unlikeCard(id)
+        .catch((err) => {
+            console.log("Error liking a card: " + err);
+    });
 }
 
 function getCardElement(data) {
     const cardSelector = "card";
     const card = new Card(data, cardSelector,
-        () => { imageModal.open(data) });
+        () => { imageModal.open(data) },
+        (event) => { openConfirmModal(event, data) },
+        likeCard, unlikeCard);
     const cardElement = card.generateCard();
+    if (data.isLiked) {
+        const likeButton = cardElement.querySelector(".card__like-button");
+        likeButton.classList.toggle("card__like-button_active");
```

Comment:
This should be done **inside** class `Card`. It's ther responsibliity of that class